src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.ts

```diff
@@ -351,39 +376,47 @@ export class DsDynamicFormControlContainerComponent extends DynamicFormControlCo
       size: 'lg'
     });
     const modalComp = this.modalRef.componentInstance;
+
+    if (hasValue(this.model.value) && !this.model.readOnly) {
+      if (typeof this.model.value === 'string') {
+        modalComp.query = this.model.value;
+      } else if (typeof this.model.value.value === 'string') {
+        modalComp.query = this.model.value.value;
+      }
+    }
+
+    if (hasValue(this.model.value)) {
+      this.model.value = '';
+      this.onChange({
+        $event: { previousIndex: 0 },
+        context: { index: 0 },
+        control: this.control,
+        model: this.model,
+        type: DynamicFormControlEventType.Change
+      });
+    }
+    this.submissionService.dispatchSave(this.model.submissionId);
+
     modalComp.repeatable = this.model.repeatable;
     modalComp.listId = this.listId;
     modalComp.relationshipOptions = this.model.relationship;
     modalComp.label = this.model.label;
     modalComp.metadataFields = this.model.metadataFields;
     modalComp.item = this.item;
     modalComp.collection = this.collection;
+    modalComp.submissionId = this.model.submissionId;
   }
 
   /**
-   * Method to move a relationship inside the list of relationships
-   * This will update the view and update the right or left place field of the relationships in the list
-   * @param event
+   * Callback for the remove event,
+   * remove the current control from its array
    */
-  moveSelection(event: CdkDragDrop<Relationship>) {
-    this.zone.runOutsideAngular(() => {
-      moveItemInArray(this.reorderables, event.previousIndex, event.currentIndex);
-      const reorderables: Reorderable[] = this.reorderables.map((reo: Reorderable, index: number) => {
-          reo.oldIndex = reo.getPlace();
-          reo.newIndex = index;
-          return reo;
-        }
-      );
-      observableCombineLatest(
-        reorderables.map((rel: ReorderableRelationship) => {
-          if (rel.oldIndex !== rel.newIndex) {
-            return this.relationshipService.updatePlace(rel);
-          } else {
-            return observableOf(undefined) as Observable<RemoteData<Relationship>>;
-          }
-        })
-      ).subscribe();
-    })
+  onRemove(): void {
+    const arrayContext: DynamicFormArrayModel = (this.context as DynamicFormArrayGroupModel).context;
+    const path = this.formBuilderService.getPath(arrayContext);
+    const formArrayControl = this.group.root.get(path) as FormArray;
+    console.log('this.listId', this.listId);
```

Comment:
Please remove this logging statement (or comment out)

Reply:
Removed console.log